cmd/influxd/upgrade/setup.go

```diff
@@ -25,81 +22,35 @@ func setupAdmin(ctx context.Context, v2 *influxDBv2, req *influxdb.OnboardingReq
 	return res, nil
 }
 
-func onboardingRequest(ui *input.UI, options *options) (*influxdb.OnboardingRequest, error) {
-	if (options.force || len(options.target.password) > 0) && len(options.target.password) < internal.MinPasswordLen {
-		return nil, internal.ErrPasswordIsTooShort
-	}
-
-	req := &influxdb.OnboardingRequest{
-		User:                   options.target.userName,
-		Password:               options.target.password,
-		Org:                    options.target.orgName,
-		Bucket:                 options.target.bucket,
-		RetentionPeriodSeconds: influxdb.InfiniteRetention,
-		Token:                  options.target.token,
-	}
-
-	if options.target.retention != "" {
-		dur, err := internal.RawDurationToTimeDuration(options.target.retention)
-		if err != nil {
-			return nil, err
-		}
-		secs, nanos := math.Modf(dur.Seconds())
-		if nanos > 0 {
-			return nil, fmt.Errorf("retention policy %q is too precise, must be divisible by 1s", options.target.retention)
-		}
-		req.RetentionPeriodSeconds = int64(secs)
-	}
-
-	if options.force {
-		return req, nil
-	}
-
-	fmt.Fprintln(ui.Writer, string(internal.PromptWithColor("Welcome to InfluxDB 2.0!", internal.ColorYellow)))
-	if req.User == "" {
-		req.User = internal.GetInput(ui, "Please type your primary username", "")
-	}
-	if req.Password == "" {
-		req.Password = internal.GetPassword(ui, false)
+func onboardingRequest(cli clients.CLI, options *options) (*influxdb.OnboardingRequest, error) {
+	setupClient := setup.Client{CLI: cli}
+	cliReq, err := setupClient.OnboardingRequest(&setup.Params{
+		Username:  options.target.userName,
+		Password:  options.target.password,
+		AuthToken: options.target.token,
+		Org:       options.target.orgName,
+		Bucket:    options.target.bucket,
+		Retention: options.target.retention,
```

Comment:
It's a bit confusing that the prompt appears to be for a different kind of string than the options value now? (nanos vs hours?)

Reply:
The help-text isn't very helpful, but I think the behavior is the same before and after this change. Whatever gets passed to `--retention` gets parsed as a `time.Duration`
* In the old impl: https://github.com/influxdata/influxdb/pull/21839/files#diff-ec11f92a3df5beb5cc1657892bfc169a49bed091ce94ebd1fae3d989a8c6eceeL43
* In the new impl: https://github.com/influxdata/influx-cli/blob/main/clients/setup/setup.go#L141

I can update the description in the help-text to show examples of durations (i.e. `72h`)